autoload/codi.vim

```diff
@@ -330,15 +330,28 @@ function! s:codi_do_update()
   " Build the command
   let cmd = s:to_list(i['bin'])
 
+  " The purpose of this is to make the REPL start from the buffer directory
+  if g:codi#use_buffer_dir
```

Comment:
Well I mostly did it for selfish reasons anyway, but I do like to share ;) Tools like this get me to the next level of productivity as a developer. Actually this is making me rethink my programming philosophy... I had named my current philosophy extreme language hacking. But then I was considering calling it leveraged language hacking, because the word extreme is overused, generic, and cheesy. But now I am realizing there is more to gaining leverage than the "language" of the code, the tools are also involved. In a sense creating libraries that give you a more powerful "language" to write you code in, and creating powerful tools, are both examples of bottom up programming from way, way down. So perhaps the lower you go, the more leverage you can create... But then it is no longer about taking it to the extreme because if you go too low, you are making your own OS and then your software will only work on your OS... Well I guess to achieve the ultimate ideal of productivity, you would have to eventually go that far. But you would have start somewhere else for practicality sake, pick up traction by going up, and only after gaining enough traction could go down further. You could install that OS on servers you run. You could use that OS when writing the code. You could still write code in languages that work on other OS's. Getting back to the word "language", you could think of the tools as a "language" we use to make software, but that would be stretching the word to make it mean what I want it to mean. So perhaps my new development philosophy is called leveraged hacking... That gets rid of the words "extreme" and "language". As I continue to ponder this, I am realizing if you went that far, you would also want to make your own language. Of course you would have to make your own interpreters / runtimes / compilers for it to run on other OSs. And then it would not be able to run on iPhone because they don't allow that kind of thing... So perhaps it would have to transpile or generate code in other languages to support all platforms. Ah, sorry for the crazy rant, it is just that it is not all that often I have these moments where my programming philosophy changes.